Checkstyle: File does not end with a newline

Hi,

I have observed some issues and improvements to be done in the GDG app, and thought to fix it up and to start contributing :)

For the same, I have imported repository directly into the android studio, but facing an issue when I try to clean/build the project.

Remarks:
1) To fix this issue TEMPORARILY, I just have commented out below line in the checkstyle.xml file
`<module name="NewlineAtEndOfFile" />`

2) I am using Windows 8.1

Here is the log:

```
Information:Gradle tasks [clean, :app:generateDebugSources, :app:generateDebugAndroidTestSources]
Warning:Dependency org.apache.httpcomponents:httpclient:4.0.1 is ignored for debug as it may be conflicting with the internal version provided by Android.
         In case of problem, please repackage it with jarjar to change the class packages
Warning:Dependency org.apache.httpcomponents:httpclient:4.0.1 is ignored for release as it may be conflicting with the internal version provided by Android.
         In case of problem, please repackage it with jarjar to change the class packages
Warning:Dependency org.apache.httpcomponents:httpclient:4.0.1 is ignored for alpha as it may be conflicting with the internal version provided by Android.
         In case of problem, please repackage it with jarjar to change the class packages
:app:clean UP-TO-DATE
:app:checkstyle
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\android\support\v4\preference\PreferenceFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\achievements\AchievementActionHandler.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\achievements\Achievements.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\activity\ArrowActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\activity\ArrowTaggedActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\activity\GdeActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\activity\GdgActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\activity\GdgNavDrawerActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\activity\ParseDeepLinkActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\activity\StartActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\activity\TrackableActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\activity\YoutubeActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\adapter\ChapterAdapter.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\adapter\ContributorAdapter.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\adapter\EventAdapter.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\adapter\GdeAdapter.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\adapter\NewsAdapter.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\ApiException.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\ApiRequest.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\GapiOkResponse.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\GapiOkTransport.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\GdeDirectory.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\GdgRequest.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\GdgStack.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\GitHub.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\JsoupRequest.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\PagedList.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\deserializer\DateTimeSerializer.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\deserializer\ZuluDateTimeDeserializer.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\Chapter.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\Contributor.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\Country.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\Directory.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\EventDetail.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\EventFullDetails.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\Gde.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\GdeList.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\GdgResponse.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\Geo.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\OrganizerCheckResponse.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\SimpleEvent.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\api\model\TaggedEvent.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\app\BackupAgent.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\app\GdgVolley.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\app\OrganizerChecker.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\crowdin\model\Translator.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\event\AgendaFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\event\EventActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\event\EventOverviewFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\event\MoreFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\eventseries\GdgEventListFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\eventseries\TaggedEventSeries.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\eventseries\TaggedEventSeriesActivity.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\eventseries\TaggedEventSeriesFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\ChangelogFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\ContributorsFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\ExtLibrariesFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\FullImageDialogFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\GdeListFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\GdgListFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\GdgRecyclerFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\GetInvolvedFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\PlainLayoutFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\SeasonsGreetingsFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\SettingsFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\SwipeRefreshRecyclerViewFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\fragment\TranslatorsFragment.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\graphics\drawable\RoundCornerDrawable.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\receiver\GCMReceiver.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\task\Builder.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\task\CommonAsyncTask.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\utils\CrashlyticsTree.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\utils\CryptoUtils.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\utils\EventDateComparator.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\utils\GingerbreadLastLocationFinder.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\utils\PrefUtils.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\utils\ScopedBus.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\utils\TaggedEventDistanceComparator.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\utils\base\ILastLocationFinder.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\view\AnimationImageView.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\view\NonSwipeableViewPager.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\view\ResizableImageView.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\view\ScrimInsetsScrollView.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\view\SlidingTabLayout.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\view\SlidingTabStrip.java:0: File does not end with a newline.
[ant:checkstyle] D:\Android\frisbee\app\src\main\java\org\gdg\frisbee\android\view\SquaredImageView.java:0: File does not end with a newline.
Error:Execution failed for task ':app:checkstyle'.
> Checkstyle rule violations were found. See the report at: file:///D:/Android/frisbee/app/build/reports/checkstyle/checkstyle.xml
Information:BUILD FAILED
Information:Total time: 30.608 secs
Information:1 error
Information:3 warnings
Information:See complete output in console
```

--- a/README.md
+++ b/README.md
@@ -29,9 +29,19 @@ Development
 
 Check out project and import it to Android Studio. 
 
-This project enforces a checkstyle. The checkstyle configuration is available in `settings/checkstyle.xml` file. Please respect to the configuration. Because of the line ending difference between Unix systems and Windows, you may get checkstyle errors. If you do, please check this [help page](https://www.jetbrains.com/idea/help/handling-lf-and-crlf-line-endings.html)
+###Checkstyle Integration
 
-Image optimization is encourged for new images. You can use the [image_optim](https://github.com/toy/image_optim) tool to do it easily. Use the following to install it. 
+This project enforces a checkstyle. The checkstyle configuration is available in `settings/checkstyle.xml` file. Please respect to the configuration. 
+
+**Note:** Because of the line ending difference between Unix systems and Windows, you may get checkstyle errors about line endings. If you do, please check this [help page](https://www.jetbrains.com/idea/help/handling-lf-and-crlf-line-endings.html).
+If you cannot resolve the issue, you can simply comment out the following line in the `checkstyle.xml` file.
+```
+<module name="NewlineAtEndOfFile" />
+```
+
+###Image Optimization
+
+Image optimization is encourged for new images. You can use the [image_optim](https://github.com/toy/image_optim) tool to do it easily. Use the following command to install it. 
 ```
 gem install image_optim image_optim_pack
 ```

